shared/packages/worker/src/worker/workers/genericWorker/expectationHandlers/lib/file.ts

```diff
@@ -206,10 +208,30 @@ export async function doFileCopyExpectation(
 				lookupSource.handle
 			)
 
-			const sourcePath = sourceHandle.fullPath
-			const targetPath = exp.workOptions.useTemporaryFilePath
-				? targetHandle.temporaryFilePath
-				: targetHandle.fullPath
+			let sourcePath: string
+			let targetPath: string
+			if (worker.agentAPI.processConfig.matchFilenamesWithoutExtension) {
```

Comment:
Should we instead use `sourceHandle.getResolvedFullPath()` here, to avoid duplicating functionality?